### PR TITLE
near protocol support

### PR DESCRIPTION
**Pull request by a10zn8**

### CODE REVIEW

File: foundation/src/main/resources/chains.yaml [hunk 1]
```diff
@@ -1268,4 +1268,32 @@ chain-settings:
...
+          code: NEAR_TESTNET
+          grpcId: 10064
+          short-names: [ near-testnet ]
+          chain-id: 0x0
+        - id: Betanet
+          priority: 1
+          code: NEAR_BETANET
+          grpcId: 10065
+          short-names: [ near-betanet ]
+          chain-id: 0x0
```
Comment (KirillPamPam) [inline foundation/src/main/resources/chains.yaml:1299]: They all have the same chain-id? Or `chain-id` just doesn't apply to near?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/DefaultNearMethods.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        "broadcast_tx_async",
+        "broadcast_tx_commit",
+    )
+
+    private val allowedMethods: Set<String> = all + add
+
+    override fun createQuorumFor(method: String): CallQuorum {
+        return when {
+            add.contains(method) -> BroadcastQuorum()
+            all.contains(method) -> AlwaysQuorum()
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/DefaultNearMethods.kt:37]: It looks like we can omit this line if we have else branch

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    override fun lowerBlockDetect(): Mono<LowerBlockData> {
+        return upstream.getIngressReader().read(JsonRpcRequest("status", listOf())).map {
+            val resp = Global.objectMapper.readValue(it.getResult(), NearStatus::class.java)
+            LowerBlockData(resp.syncInfo.earliestHeight, null, resp.syncInfo.earliestBlockTime.toEpochMilli())
+        }
+    }
+
+    override fun periodRequest(): Long {
+        return 120
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt:23]: Maybe decrease period if near has super cool method with `earliestHeight`? Like in solana

File: foundation/src/main/resources/chains.yaml (hunk 1 above)
Comment (a10zn8) [reply foundation/src/main/resources/chains.yaml:1299]: it applies only on ETH networks, solana, gear and near doesn't have same value.
Sometimes they have some similar value, but it is not a hex string. We have to investigate it later I think

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt (hunk 3 above)
Comment (a10zn8) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt:23]: Ok! It was just copypaste from starknet. 
Which value is preferable?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/DefaultNearMethods.kt (hunk 2 above)
Comment (a10zn8) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/DefaultNearMethods.kt:37]: will fix

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt (hunk 3 above)
Comment (KirillPamPam) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt:23]: I suggest 3 (minutes), like in solana

**Pull request merged**